reflex/custom_components/custom_components.py

```diff
@@ -70,6 +71,7 @@ def _create_package_config(module_name: str, package_name: str):
         )
 
 
+@lru_cache(maxsize=None)
 def _get_package_config(exit_on_fail: bool = True) -> dict:
```

Comment:
In `_validate_project_info` we interactively ask for info, and may end up updating the `pyproject.toml`, making this caching invalid - since afterwards we try to get the toml in `_collect_details_for_gallery`.

I'm thinking it may be okay as is originally - the read operations should be pretty fast, and will ensure we're always reading the up to date config. What do you think?

Reply:
Possibly. Probably the best thing to do next is write up some unit tests for this module. I do feel a bit uncomfortable changing code that I only tested manually.

Reply:
The pyproject.toml file is redundantly loaded at least 3 times with no changes until the caching would be invalidated. This code could probably use some heavy refactoring, but I'll leave it for now.